app/controllers/api/v1/merchants/revenue_controller.rb

```diff
@@ -1,26 +1,23 @@
 class Api::V1::Merchants::RevenueController < ApplicationController
-  include ActionView::Helpers::NumberHelper  
-  respond_to :json
-  
+
   def index
-    @merchants = Merchant.order_by_revenue(params[:quantity].to_i)
+    @merchants = Merchant.order_by_revenue(params[:quantity])
     respond_with @merchants
   end
-  
+
   def show
     merchant = Merchant.find(params[:id])
     if params[:date]
-      @revenue = merchant.revenue_on(params[:date].to_date)
+      @revenue = merchant.revenue_on(params[:date])
```

Comment:
**`.to_date` was causing the spec harness test to fail, but the rspec test(s) to pass. I re-worked the rspec test(s) to pass in the date in the same format the spec harness uses.**